app/Actions/Album/Delete.php

```diff
@@ -78,18 +78,23 @@ public function do(array $albumIDs): FileDeleter
 			// loading expensive `min_taken_at` and `max_taken_at`.
 			$albums = Album::query()
 				->without(['cover', 'thumb'])
-				->select(['id', 'parent_id', '_lft', '_rgt'])
+				->select(['id', 'parent_id', '_lft', '_rgt', 'track_short_path'])
 				->findMany($albumIDs);
 
+			$recursiveAlbums = $albums;
+
 			/** @var Album $album */
 			foreach ($albums as $album) {
-				// Collect the IDs of all (aka recursive) sub-albums in each album
-				$recursiveAlbumIDs = array_merge($recursiveAlbumIDs, $album->descendants()->pluck('id')->all());
+				// Collect all (aka recursive) sub-albums in each album
+				$subAlbums = $album->descendants()->select(['id', 'track_short_path'])->all();
+				$recursiveAlbumIDs = array_merge($recursiveAlbumIDs, $subAlbums->pluck('id')->all());
+				$recursiveAlbums = array_merge($recursiveAlbums, $subAlbums);
 			}
 
 			// Delete the photos from DB and obtain the list of files which need
 			// to be deleted later
 			$fileDeleter = (new PhotoDelete())->do($unsortedPhotoIDs, $recursiveAlbumIDs);
+			$fileDeleter->addRegularFiles($recursiveAlbums->pluck('track_short_path'));
```

Comment:
Nearly there! My suggestions avoid some unnecessary memory consumption and copying.

```suggestion
			$fileDeleter->addRegularFiles($albums->pluck('track_short_path'));
			$fileDeleter->addRegularFiles($subAlbums->pluck('track_short_path'));
```

Reply:
But `subAlbums` is a variable which is set every time in the `foreach` statement in line 89 and only contains sub albums of one top-level album. That's the reason for this new `recursiveAlbums` variable.

Reply:
My fault! :-(

> That's the reason for this new `recursiveAlbums` variable.

Understood. But maybe we could just change it into a `recursiveAlbumTracks` in analogy to `recursiveAlbumIDs`. I would like to avoid duplicating entire `Album` models.

Reply:
https://github.com/LycheeOrg/Lychee/pull/1274/commits/37f013e88c3b7d62b7d912cb21336b201052a85f